homeassistant/components/prometheus/__init__.py

```diff
@@ -355,13 +362,21 @@ def state_as_number(state: State) -> float:
             value = 0
         return value
 
-    @staticmethod
-    def _labels(state: State) -> dict[str, Any]:
-        return {
+    def _labels(self, state: State) -> dict[str, Any]:
+        final_labels = {
             "entity": state.entity_id,
             "domain": state.domain,
             "friendly_name": state.attributes.get(ATTR_FRIENDLY_NAME),
         }
+        if self._include_extra_labels:
+            final_labels.update(
+                {
+                    "area": state.attributes.get(ATTR_AREA_ID) or "",
```

Comment:
Note that we'd need to handle changes to these attributes at runtime, calling `_remove_labelsets` to clean up the old metrics. We already do it for `ATTR_FRIENDLY_NAME`, and will need to make that code more generic to handle these too:

https://github.com/home-assistant/core/blob/9c026bc442ed03427e9581cb2579411abfe6f006/homeassistant/components/prometheus/__init__.py#L211-L214

Reply:
This makes sense. Probably wise for me to try and write a test aroudn this too